generator/feature_sorter.cpp

```diff
@@ -162,19 +164,23 @@ namespace feature
 
       {
         FileWriter w = m_writer.GetWriter(METADATA_INDEX_FILE_TAG);
-        for (auto const & v : m_MetadataIndex)
+        for (auto const & v : m_metadataIndex)
         {
           WriteToSink(w, v.key);
           WriteToSink(w, v.value);
         }
       }
 
-      m_MetadataWriter->Flush();
-      m_writer.Write(m_MetadataWriter->GetName(), METADATA_FILE_TAG);
+      m_metadataWriter->Flush();
+      m_writer.Write(m_metadataWriter->GetName(), METADATA_FILE_TAG);
+
+      m_searchTokensWriter->Flush();
+      m_writer.Write(m_searchTokensWriter->GetName(), SEARCH_TOKENS_FILE_TAG);
 
       m_writer.Finish();
 
-      FileWriter::DeleteFileX(m_MetadataWriter->GetName());
+      FileWriter::DeleteFileX(m_metadataWriter->GetName());
```

Comment:
In case of exceptions it's better to declare a couple of MY_SCOPE_GUARDS at the top of the constructor.